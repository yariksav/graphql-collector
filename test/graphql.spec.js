
const { splitter, collect } = require('../src/')
const path = require('path')
const should = require('should')

const moksDir = path.join(__dirname, 'mocks')

describe('serach and join schemas', () => {

  it ('should join all graphql files', async () => {
    const res = await collect(moksDir)
    should(res).be.Object().and.have.properties(['changePassword', 'getProfile', 'login', 'register'])
    expect(res).toMatchSnapshot()
  })

  it ('should join and collect movie files', async () => {
    const res = await collect(moksDir + '/movie')
    should(res).be.Object().and.have.properties(['ListMovies'])
    expect(res).toMatchSnapshot()
  })

  it ('should return empty object when no one file was found', async () => {
    const res = await collect(moksDir + '/empty')
    should(res).be.Object().and.empty()
  })

  it ('should join all graphql files', async () => {
    try {
      await collect(moksDir + '/nonexists')
      should(true).be.false()
    } catch (e) {
      should(e.message).startWith('ENOENT: no such file or directory')
    }
  })

  it ('should return', async () => {
    const res = await collect(moksDir + '/user')
    expect(res).toMatchSnapshot()
  })
})
