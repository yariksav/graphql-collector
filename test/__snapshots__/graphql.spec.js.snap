// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serach and join schemas should join all graphql files 1`] = `
Object {
  "ListMovies": "query ListMovies { allFilms { films { ...Movie } } } fragment Movie on Film { id title director planetConnection { planets { ...Place } } } fragment Place on Planet { name climates }",
  "changePassword": "mutation changePassword($oldPassword: String, $password: String!) { changePassword(oldPassword: $oldPassword, password: $password) }",
  "findUsers": "query findUsers($filter: FilterInput) { findUsers(filter: $filter) { ...userFields } } fragment userFields on User { id username email firstname lastname fullname avatar { ...avatarFields } account_status options } fragment avatarFields on Avatar { large original thumbnail normal small }",
  "getProfile": "query getProfile { profile { ...userFields } } fragment userFields on User { id username email firstname lastname fullname avatar { ...avatarFields } account_status options } fragment avatarFields on Avatar { large original thumbnail normal small }",
  "login": "mutation login($uid: String!, $password: String!) { login(uid: $uid, password: $password) { ...jwtToken } } fragment jwtToken on JwtToken { token refreshToken type }",
  "register": "mutation register($username: String, $email: String!, $password: String!, $firstname: String, $lastname: String) { register(username: $username, email: $email, password: $password, firstname: $firstname, lastname: $lastname) { ...jwtToken } } fragment jwtToken on JwtToken { token refreshToken type }",
}
`;

exports[`serach and join schemas should join and collect movie files 1`] = `
Object {
  "ListMovies": "query ListMovies { allFilms { films { ...Movie } } } fragment Movie on Film { id title director planetConnection { planets { ...Place } } } fragment Place on Planet { name climates }",
}
`;

exports[`serach and join schemas should return 1`] = `
Object {
  "changePassword": "mutation changePassword($oldPassword: String, $password: String!) { changePassword(oldPassword: $oldPassword, password: $password) }",
  "findUsers": "query findUsers($filter: FilterInput) { findUsers(filter: $filter) { ...userFields } } fragment userFields on User { id username email firstname lastname fullname avatar { ...avatarFields } account_status options } fragment avatarFields on Avatar { large original thumbnail normal small }",
  "getProfile": "query getProfile { profile { ...userFields } } fragment userFields on User { id username email firstname lastname fullname avatar { ...avatarFields } account_status options } fragment avatarFields on Avatar { large original thumbnail normal small }",
  "login": "mutation login($uid: String!, $password: String!) { login(uid: $uid, password: $password) { ...jwtToken } } fragment jwtToken on JwtToken { token refreshToken type }",
  "register": "mutation register($username: String, $email: String!, $password: String!, $firstname: String, $lastname: String) { register(username: $username, email: $email, password: $password, firstname: $firstname, lastname: $lastname) { ...jwtToken } } fragment jwtToken on JwtToken { token refreshToken type }",
}
`;
